static/examples/audio.html

```diff
@@ -0,0 +1,188 @@
+          }
+          return curve;
+        };
+
+
+        if (navigator.mediaDevices.getUserMedia) {
+           const constraints = {
+            audio: {
+              echoCancellation: true,
+              noiseSupression: false,
+              volume: {
+                exact: {
+                  min: 0,
+                  max: 90,
+                },
+                ideal: {
+                  min: 0,
+                  max: 90,
+                }
+              }
+            }
+          }
+          navigator.mediaDevices.getUserMedia(constraints)
+            .then(function(stream) {
+              source = audioCtx.createMediaStreamSource(stream);
+              source.connect(distortion);
+              distortion.connect(biquadFilter);
+              biquadFilter.connect(gainNode);
+              gainNode.connect(delay);
+              delay.connect(audioCtx.destination);
+
+              range.oninput = function() {
+                gainNode.gain.value = range.value;
+              }
+              // Delay time needs to between 0, 1 but the slider only returns integers
+              delayrange.oninput = function() {
+                delay.delayTime.value = delayrange.value / 10;
+              }
+              voiceChange();
+            })
+            .catch( function(err) { console.log('The following gUM error occured: ' + err);})
+        } else {
+          console.log('getUserMedia not supported on your browser!');
+        }
+
+        function voiceChange() {
+
+          distortion.oversample = '4x';
+          biquadFilter.gain.setTargetAtTime(0, audioCtx.currentTime, 0)
+
+          let voiceSetting = voiceSelect.value;
+
+            biquadFilter.disconnect(0);
+            biquadFilter.connect(gainNode);
+
+          if(voiceSetting == "distortion") {
+            distortion.curve = makeDistortionCurve(400);
+            biquadFilter.disconnect(0);
+            biquadFilter.connect(gainNode);
+          } else if(voiceSetting == "biquad") {
+            biquadFilter.type = "lowshelf";
+            biquadFilter.frequency.setTargetAtTime(1000, audioCtx.currentTime, 0)
+            biquadFilter.gain.setTargetAtTime(25, audioCtx.currentTime, 0)
+          }
+        }
+
+        const mute = document.getElementById('mute');
+        voiceSelect.onchange = function() {
+          voiceChange();
+        };
+
+        mute.onclick = voiceMute;
+
+        function voiceMute() {
+          if(mute.id === "") {
+            gainNode.gain.value = 0;
+            mute.id = "activated";
+            mute.innerHTML = "Unmute";
+          } else {
+            gainNode.gain.value = 1;
+            mute.id = "";
+            mute.innerHTML = "Mute";
+          }
+        }
+      }
+    </script>
+  </head>
+  <body>
+    <input type="button" value="Start Microphone" onclick="start()"/>
+    <canvas class="visualizer" width="640" height="100"></canvas> 
```

Comment:
I assume that this visualization code is not yet implemented, but looking forward to it.